src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -114,6 +118,11 @@ class FunSpec private constructor(builder: Builder) {
       codeWriter.emitCode(": %T", returnType)
     }
 
+    if (delegateConstructorParameters.isNotEmpty()) {
+      codeWriter.emitCode(delegateConstructorParameters
+          .joinToCode(prefix = " : this(", suffix = ")"))
```

Comment:
What about calling `super`?

Reply:
I can extend the API to allow for specifying which constructor to delegate to, `super` or `this`.